gazebo_ros/src/gazebo_ros_api_plugin.cpp

```diff
@@ -183,6 +183,8 @@ void GazeboRosApiPlugin::loadGazeboRosApiPlugin(std::string world_name)
   gazebonode_->Init(world_name);
   //stat_sub_ = gazebonode_->Subscribe("~/world_stats", &GazeboRosApiPlugin::publishSimTime, this); // TODO: does not work in server plugin?
   factory_pub_ = gazebonode_->Advertise<gazebo::msgs::Factory>("~/factory");
+  factory_light_pub_ = gazebonode_->Advertise<gazebo::msgs::Light>("~/factory/light");
```

Comment:
This is actually made for matching the Gazebo communication channel's name, as it's happening for all the other publishers. But if you think light_create_pub_ is a better idea I will change it.